rider/src/main/kotlin/com/jetbrains/rider/plugins/unity/notifications/OpenUnityProjectAsFolderNotification.kt

```diff
@@ -32,26 +35,33 @@ class OpenUnityProjectAsFolderNotification(project: Project) : ProtocolSubscribe
     }
 
     init {
-        project.solution.frontendBackendModel.unityApplicationData.advise(projectComponentLifetime) {
+        project.solution.isLoaded.whenTrue(projectComponentLifetime) {
             if (!UnityProjectDiscoverer.getInstance(project).isUnityProjectFolder)
-                return@advise
+                return@whenTrue
 
             val solutionDescription = project.solutionDescription
-            val title = "Unity features unavailable"
-            val content = "Make sure <b>Rider package</b> is installed in Unity’s Package Manager and  Rider is set as the External Editor. Reopen Rider from Unity."
-            if (solutionDescription is RdExistingSolution){ // proper solution
-                if (UnityInstallationFinder.getInstance(project).requiresRiderPackage() && !PackageManager.getInstance(project).hasPackage("com.unity.ide.rider")){
-                    val notification = Notification(notificationGroupId.displayId, title, content, NotificationType.WARNING)
-                    Notifications.Bus.notify(notification, project)
-                }
+            val title = "Advanced Unity integration is unavailable"
+            var content = "Make sure <b>Rider package</b> is installed in Unity’s Package Manager and Rider is set as the External Editor."
+            if (solutionDescription is RdExistingSolution) { // proper solution
+                it.startNonUrgentBackgroundAsync {
+                    // Sometimes in Unity "External Script Editor" is set to "Open by file extension"
+                    // We check that Library/EditorInstance.json is present, but protocol connection was not initialized within 1 second.
+                    delay(1000)
```

Comment:
I don't like the arbitrary 1 second here. Does the `isLoaded` property only fire once the whole solution has finished everything (initial file processing, cache loading, etc.)? So this is just an extra delay?